app/src/main/java/com/example/crearrepositorio/ui/viewModel/MainViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.crearrepositorio.ui.viewModel
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+
+class MainViewModel : ViewModel(){
+
+    private val _estados = MutableStateFlow<UiState>(UiState.Idle)
+    val stateHome : StateFlow<UiState> = _estados.asStateFlow()
+
+    fun navigateToFilms(){
+        _estados.update { currentState ->
```

Comment:
Si no me equivoco el compilador os dirá que, como no estáis usando el currenState de la lambda, podéis eliminarlo para que quede más limpio.

```suggestion
        _estados.update {  UiState.NavigateToFilms }